src/MobileUI/PopupPages/RedeemReward.xaml

```diff
@@ -0,0 +1,257 @@
+                           Grid.Column="0"
+                           Text="Product Cost"
+                           FontSize="18"
+                           HorizontalOptions="Center"
+                           VerticalOptions="Center"/>
+                    
+                    <Border Grid.Row="1"
+                            Grid.Column="1"
+                            BackgroundColor="{StaticResource FlyoutBackgroundColour}"
+                            StrokeThickness="0"
+                            Padding="15,4">
+                        <Border.StrokeShape>
+                            <RoundRectangle CornerRadius="6" />
+                        </Border.StrokeShape>
+                        <HorizontalStackLayout Spacing="8">
+                            <Label 
+                                TextColor="{StaticResource Coin}"
+                                Text="&#xf51e;"
+                                FontFamily="FA6Solid" 
+                                VerticalOptions="Center"
+                                FontSize="11" 
+                                InputTransparent="True" />
+                            <Label Text="{Binding Cost, StringFormat='{0:n0}'}"
+                                   VerticalOptions="Center"
+                                   FontSize="18"/>
+                        </HorizontalStackLayout>
+                    </Border>
+                </Grid>
+                
+                <!-- Address section -->
+                <VerticalStackLayout IsVisible="{Binding IsAddressVisible}"
+                                     Spacing="2"
+                                     Margin="0,0,0,30">
+                    <Label Text="Address Line 1"
+                           FontSize="18"
+                           TextColor="{StaticResource Gray300}"/>
+                    <Entry Text="{Binding AddressLine1}"
+                           Margin="0,0,0,8"/>
+                    
+                    <Label Text="Address Line 2"
+                           FontSize="18"
+                           TextColor="{StaticResource Gray300}"/>
+                    <Entry Text="{Binding AddressLine2}"
+                           Margin="0,0,0,8"/>
+                    
+                    <Label Text="Suburb"
+                           FontSize="18"
+                           TextColor="{StaticResource Gray300}"/>
+                    <Entry Text="{Binding AddressSuburb}"
+                           Margin="0,0,0,8"/>
+                    
+                    <Label Text="Postcode"
+                           FontSize="18"
+                           TextColor="{StaticResource Gray300}"/>
+                    <Entry Text="{Binding AddressPostcode}"
+                           Margin="0,0,0,8"/>
+                    
+                    <FlexLayout BindableLayout.ItemsSource="{Binding States}"
+                                Wrap="Wrap">
+                        <BindableLayout.ItemTemplate>
```

Comment:
This UI is a bit clunky. Can we ask the designers to review this? We might be able to improve it in a future iteration.